packages/grafana-ui/src/components/Slider/Slider.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { FunctionComponent } from 'react';
+import { Range, createSliderWithTooltip } from 'rc-slider';
+import { css } from 'emotion';
+import { stylesFactory, withTheme } from '../../themes';
+import { GrafanaTheme } from '@grafana/data';
+import { Themeable } from '../../types';
+
+export interface Props extends Themeable {
+  min: number;
+  max: number;
+  orientation: 'horizontal' | 'vertical';
+  reverse?: boolean;
+  size?: string;
+  formatTooltipResult?: (value: any) => number | string;
+  onChange?: (value: any) => void;
+}
+
+const getStyles = stylesFactory((theme: GrafanaTheme, orientation: 'horizontal' | 'vertical', size: string) => {
+  const container =
+    orientation === 'horizontal'
+      ? css`
+          width: ${size};
+          margin: 20px;
+        `
+      : css`
+          height: ${size};
+          margin: 20px;
+        `;
+  return {
+    container: container,
+  };
+});
+
+const UnThemedSlider: FunctionComponent<Props> = ({
+  theme,
+  min,
+  max,
+  onChange,
+  orientation,
+  size,
+  reverse,
+  formatTooltipResult,
+}) => {
+  const toltipSize = size || '200px';
+  const isHorizontal = orientation === 'horizontal';
+  const styles = getStyles(theme, orientation, toltipSize);
+  const RangeWithTooltip = createSliderWithTooltip(Range);
+  return (
+    <div className={styles.container}>
+      <RangeWithTooltip
+        tipProps={{ visible: true, placement: isHorizontal ? 'top' : 'right' }}
+        min={min}
+        max={max}
+        defaultValue={[min, max]}
+        tipFormatter={(value: string | number) => (formatTooltipResult ? formatTooltipResult(value) : value)}
+        onChange={onChange}
+        vertical={!isHorizontal}
+        reverse={reverse}
+      />
+    </div>
+  );
+};
+
+export const Slider = withTheme(UnThemedSlider);
```

Comment:
For functional components do not use withTheme, you can use the useTheme() hook instead inside the function component, example: https://github.com/grafana/grafana/blob/eba0390502f39276f657f21e4270b2b2008c1df0/packages/grafana-ui/src/components/Forms/Select/SelectMenu.tsx#L15